### PR TITLE
CIV-10370 Make available Claimant Lip

### PR DESCRIPTION
**Pull request by miguelMolina3691**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-10370

### Change description ###
ClaimantDocuments

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by LeonardoPalmeiro — CHANGES_REQUESTED** [review]
if possible increase coverage https://sonarcloud.io/summary/new_code?id=civil-citizen-ui&pullRequest=2282
because the new methods that we're distinguish if is claimant or defendant is not being test>
image

File: src/main/routes/features/caseProgression/typeOfDocumentsController.ts [hunk 1]
```diff
@@ -34,7 +40,8 @@ typeOfDocumentsController.get(TYPES_OF_DOCUMENTS_URL,
   (async (req: AppRequest, res: Response, next: NextFunction) => {
     try {
       const claimId = req.params.id;
-      const documentsList = await getDocuments(claimId,ClaimantOrDefendant.DEFENDANT);
+      const claim = await getCaseDataFromStore(claimId);
```
Comment (LeonardoPalmeiro) [inline src/main/routes/features/caseProgression/typeOfDocumentsController.ts:43]: inside the **_await getDocument_**s we're getting const caseData = await getCaseDataFromStore(claimId); as well.
do we really need to fetch twice?

File: src/main/routes/features/caseProgression/typeOfDocumentsController.ts [hunk 2]
```diff
@@ -45,15 +52,16 @@ typeOfDocumentsController.get(TYPES_OF_DOCUMENTS_URL,
...
     const typeDocumentList= getTypeDocumentForm(req);
     const form = new GenericForm(typeDocumentList);
 
     form.validateSync();
     if (form.hasErrors()) {
       await renderView(res, claimId,form);
     } else {
-      await saveCaseProgression(claimId, form.model, dqPropertyName);
-      await deleteUntickedDocumentsFromStore(claimId, ClaimantOrDefendant.DEFENDANT);
+      await saveCaseProgression(claimId, form.model, claim.isClaimant() ? dqPropertyNameClaimant : dqPropertyName);
```
Comment (LeonardoPalmeiro) [inline src/main/routes/features/caseProgression/typeOfDocumentsController.ts:63]: My suggestion in here is create a variable before then we don't need to access all the time claim.isClaimant() will be much more clear :)

File: src/main/routes/features/caseProgression/uploadDocumentsController.ts [hunk 3]
```diff
@@ -60,13 +62,15 @@ uploadDocumentsController.get(CP_UPLOAD_DOCUMENTS_URL, (async (req: Request, res
 uploadDocumentsController.post(CP_UPLOAD_DOCUMENTS_URL, (async (req, res, next) => {
   try {
     const claimId = req.params.id;
+    const claim: Claim = await getCaseDataFromStore(claimId);
```
Comment (LeonardoPalmeiro) [inline src/main/routes/features/caseProgression/uploadDocumentsController.ts:65]: in **_await renderView(res, claimId, form)_**; we are making another call to the data base, it is really necessary ?

File: src/main/services/features/caseProgression/disclosureService.ts [hunk 4]
```diff
@@ -9,15 +9,24 @@ import {
   FileOnlySection, TypeOfDocumentSection,
   UploadDocumentsUserForm,
 } from 'models/caseProgression/uploadDocumentsUserForm';
+import {UploadDocuments} from 'models/caseProgression/uploadDocumentsType';
 
 export const getDisclosureContent = (claim: Claim, form: GenericForm<UploadDocumentsUserForm>): ClaimSummaryContent[][] => {
   const sectionContent = [];
 
-  if (claim.caseProgression?.defendantUploadDocuments?.disclosure[0]?.selected) {
+  let uploadDocuments: UploadDocuments;
```
Comment (LeonardoPalmeiro) [inline src/main/services/features/caseProgression/disclosureService.ts:17]: my suggestion in here is
const uploadDocuments: UploadDocuments = claim.isClaimant() : claim.caseProgression?.claimantUploadDocuments ? claim.caseProgression?.defendantUploadDocuments;

**Review by m-meulendijks-v1 — DISMISSED** [review]

File: src/main/routes/features/caseProgression/typeOfDocumentsController.ts (hunk 2 above)
Comment (miguelMolina3691) [reply src/main/routes/features/caseProgression/typeOfDocumentsController.ts]: Modified

File: src/main/routes/features/caseProgression/typeOfDocumentsController.ts (hunk 1 above)
Comment (miguelMolina3691) [reply src/main/routes/features/caseProgression/typeOfDocumentsController.ts]: Solved

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by m-meulendijks-v1 — DISMISSED** [review]

**Review by laarmada — DISMISSED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

File: src/main/routes/features/caseProgression/documentsUploadedController.ts [hunk 5]
```diff
@@ -6,15 +6,19 @@ import {
...
 const uploadDocumentsViewPath = 'features/caseProgression/documents-uploaded';
 const documentsUploadedController = Router();
+const civilServiceApiBaseUrl = config.get<string>('services.civilService.url');
+const civilServiceClient: CivilServiceClient = new CivilServiceClient(civilServiceApiBaseUrl);
 
 documentsUploadedController.get(CP_EVIDENCE_UPLOAD_SUBMISSION_URL, (async (req:Request, res:Response, next: NextFunction) => {
   try {
     const claimId = req.params.id;
-    const claim = await getCaseDataFromStore(claimId);
+    const claim = await civilServiceClient.retrieveClaimDetails(claimId, <AppRequest>req);
```
[the remark this answers is not in the document]
Comment (miguelMolina3691) [reply src/main/routes/features/caseProgression/documentsUploadedController.ts:21]: as no further code was needed, I thought I didn't need to create a service to call the client, in other controllers it is done in the same way

**Review by dharmendrak — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Pull request merged**